### PR TITLE
Source Control sample

### PR DESCRIPTION
**Pull request by jan-dolejsi**
This sample implements a minimal source control provider. It shows how the source control experience in VS Code could be used to for example interact with html/js/css code stored and versioned in JSFiddle.

### CODE REVIEW

**Comment by jan-dolejsi** [comment]
Thanks for reviewing and feedback. I will retest the _discard_ and _commit_ in another environment. Close and re-open local files is a scenario that I missed. Will fix it.

**Comment by jan-dolejsi** [comment]
The sample is refactored to fix the third issue and to support multi-root workspaces. If you still find the _Discard local changes_ and _Commit local changes_ not working, please help me with a repro scenario.

**Comment by joaomoreno** [comment]
I still see third issue.

1. Run `Open JSFiddle`. Accept default value. Accept to clear the workspace.
2. Close all 3 buffers.
3. Open each file from the Explorer view. Make changes in all.

At first it seemed that it picked up changes in the CSS file and none in the other two files. After undoing the changes in the CSS, it continued to tell me it has changes, even though nothing has changed:

image

---

Also, when I run `Open JSFiddle` a second time, the following error comes up:

```
Error: command 'extension.source-control.refresh' already exists
extension.js:54
	at d.registerCommand (/snap/code-insiders/91/usr/share/code-insiders/resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:466:33)
	at Object.registerCommand (/snap/code-insiders/91/usr/share/code-insiders/resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:678:224)
	at new FiddleSourceControl (/home/joao/Work/vscode-extension-samples/source-control-sample/out/fiddleSourceControl.js:29:52)
	at Function.<anonymous> (/home/joao/Work/vscode-extension-samples/source-control-sample/out/fiddleSourceControl.js:52:20)
	at Generator.next (<anonymous>)
	at fulfilled (/home/joao/Work/vscode-extension-samples/source-control-sample/out/fiddleSourceControl.js:4:58)
```

**Comment by jan-dolejsi** [comment]
Joao, the stack-track indicates that you are not testing the latest codebase. The command `extension.source-control.refresh` handler is no longer registered by `FiddleSourceControl` in the code I pushed yesterday. It is registered in the `extension.ts` inside the `activate` function to avoid exactly the error you observed.

**Comment by joaomoreno** [comment]
Oh I might've forgotten to compile. Sorry for that. Let me try again.

**Comment by joaomoreno** [comment]
When I try to "commit", I get the following:

```
  ERR The page responsed with a status code different than 200: 422
```

**Comment by jan-dolejsi** [comment]
I see. It turns out the library cannot write back to JSFiddle anymore. The `demo` mock fiddle works, but real fiddle cannot write back. I added an option to open the fiddle in the browser instead : (
How is that for a sample?

**Comment by joaomoreno** [comment]
That seems to work OK! Thanks @jan-dolejsi! :fireworks:
